### PR TITLE
Misc README cleanup

### PR DESCRIPTION
**Pull request by sleepinyourhat**
The README is still a bit messy, but I just took a quick pass to resolve some obvious issues.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,28 +1,43 @@
...
-The 'j' in `jiant` stands for JSALT. That's all the acronym we have.
+`jiant` is a work-in-progress software toolkit for natural language processing research, designed to facilitate work on multitask learning and transfer learning for sentence understanding tasks.
 
-jiant has been used in these two papers so far:
+A few things you might want to know about `jiant`:
 
-- [Looking for ELMo's Friends: Sentence-Level Pretraining Beyond Language Modeling](https://arxiv.org/abs/1812.10860)
-- [What do you learn from context? Probing for sentence structure in contextualized word representations](https://openreview.net/forum?id=SJzSgnRcKX) ("Edge Probing")
+- `jiant` is configuration-driven. You can run an enormous variety of experiments by simply writing configuration files. Of course, if you need to add any major new features, you can also easily edit or extend the code.
+- `jiant` contains implementations of strong baselines for the [GLUE](https://gluebenchmark.com) and [SuperGLUE](https://gluebenchmark.com/super) benchmarks, and it's the recommended starting point for work on these benchmarks.
```
Comment (W4ngatang) [inline README.md]: super.gluebenchmark.com

File: README.md [hunk 2]
```diff
@@ -37,22 +52,10 @@ You will also need to install dependencies for nltk if you do not already have t
...
-
-If you already cloned and just need to get the submodules, you can do:
-```sh
-git submodule update --init --recursive
-```
-
-## Downloading data
-
-The repo contains a convenience python script for downloading all [GLUE](https://www.nyu.edu/projects/bowman/glue.pdf) data and standard splits.
+The repo contains a convenience python script for downloading all [GLUE](https://gluebenchmark.com/) data and standard splits.
```
Comment (W4ngatang) [inline README.md]: small thing, but previously this script was in a git gist, and that was the main version I updated. It'd be nice to have it in a single place, either this repo or a gist, so we don't have to maintain two versions.

File: README.md [hunk 3]
```diff
@@ -152,16 +152,16 @@ To use the ON-LSTM sentence encoder from [Ordered Neurons: Integrating Tree Stru
...
+When using BERT, we follow the procedures set out in the original work as closely as possible: For pair sentence tasks, we concatenate the sentences with a special `[SEP]` token. Rather than max-pooling, we take the first representation of the sequence (corresponding to the special `[CLS]` token) as the representation of the entire sequence.
 We also have support for the version of Adam that was used in training BERT (``optimizer = bert_adam``).
 
+We also include an experimental option to use a shared [Transformer](https://arxiv.org/abs/1706.03762) in place of the shared BiLSTM by setting ``sent_enc = transformer``. When using a Transformer, we use the [Noam learning rate scheduler](https://github.com/allenai/allennlp/blob/master/allennlp/training/learning_rate_schedulers.py#L84), as that seems important to training the Transformer thoroughly.
```
Comment (W4ngatang) [inline README.md]: dumb way to support Transformers easily is to allow for resetting the BERT and GPT weights?

File: README.md [hunk 4]
```diff
@@ -193,7 +193,7 @@ To add new tasks, you should:
 
 1. Add your data to the ``data_dir`` you intend to use. When constructing your task class (see next bullet), make sure you specify the correct subfolder containing your data.
 
-2. Shuffle your training and validation data if there are many rows (>10k) to avoid training artefacts. Indeed, jiant loads 10k examples at a time in memory and then only it shuffles them. This could create issues if your data is sorted, e.g: If your data is sorted by label, the model would go through all the examples of a class before finding one of another, so it would learn to always predict the first class. If you reach 100% accuracy before one epoch, this is likely the case.
+2. Shuffle your training and validation data if there are many rows (>10k) to avoid training artefacts. Indeed, `jiant` loads 10k examples at a time in memory and then only it shuffles them. This could create issues if your data is sorted, e.g: If your data is sorted by label, the model would go through all the examples of a class before finding one of another, so it would learn to always predict the first class. If you reach 100% accuracy before one epoch, this is likely the case.
```
Comment (W4ngatang) [inline README.md]: artefacts -> artifacts? Or suddenly British?

**Comment by pruksmhc** [comment]
To add, here's the task tutorial: 
http://socrates.io/#neunXWc 
and "Get Started" coming soon!

File: README.md (hunk 3 above)
Comment (sleepinyourhat) [reply README.md]: 🤷‍♂

**Comment by sleepinyourhat** [comment]
This is just meant to be an initial pass of cleanup, to improve this content on the assumption that most of it will survive in some form.

Once the tutorials are done, @pruksmhc, mind making a PR shifting any redundant material out of README and adding tutorial pointers?

**Pull request merged**